recipes/tbb/all/conanfile.py

```diff
@@ -103,22 +102,62 @@ def add_flag(name, value):
 MALLOC_NO_VERSION.DLL =
 MALLOCPROXY.DLL =
 MALLOCPROXY.DEF =
-""")
+""".format("xilib" if self.settings.compiler == "intel" else "lib"))
             extra = "" if self.options.shared else "extra_inc=big_iron_msvc.inc"
         else:
             extra = "" if self.options.shared else "extra_inc=big_iron.inc"
-        arch = {"x86": "ia32",
-                "x86_64": "intel64",
-                "armv7":  "armv7",
-                "armv8": "aarch64"}.get(str(self.settings.arch))
-        if self.settings.compiler in ['gcc', 'clang', 'apple-clang']:
-            if str(self.settings.compiler.libcxx) in ['libstdc++', 'libstdc++11']:
+
+        arch = {
+            "x86": "ia32",
+            "x86_64": "intel64",
+            "armv7": "armv7",
+            "armv8": "aarch64",
+        }[str(self.settings.arch)]
```

Comment:
Why? I rather get an exception here, rather than mysteriously having a `arch=None` in the make command, causing more cryptic errors.

Reply:
So add specific treatment for IndexError.

Reply:
Sure, I think it is reasonable to have in the configure method a check for one of the supported archs, but I don't understand the general suggestion to use get instead of indexing. Is there some style guideline I'm not aware of, or just your personal preference?

Reply:
personal opinion and most used pattern in CCI.

Reply:
Given that it is personal opinion and there is no formalized style guidelines in this regard, I'm going to leave it as is.